{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Matching"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Purpose:\n",
    "\n",
    "* Find exact and fuzzy overlaps between significant variants that are (e/p/m)QTLs\n",
    "* Create organized tables relating variants to relevant biological features"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load in Data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>trait</th>\n",
       "      <th>stratum</th>\n",
       "      <th>phenotype.id</th>\n",
       "      <th>gene.name</th>\n",
       "      <th>start_pos</th>\n",
       "      <th>end_pos</th>\n",
       "      <th>strand</th>\n",
       "      <th>nsnps</th>\n",
       "      <th>variant.id1</th>\n",
       "      <th>variant.id2</th>\n",
       "      <th>PP.H0.abf.Exam1</th>\n",
       "      <th>PP.H1.abf.Exam1</th>\n",
       "      <th>PP.H2.abf.Exam1</th>\n",
       "      <th>PP.H3.abf.Exam1</th>\n",
       "      <th>PP.H4.abf.Exam1</th>\n",
       "      <th>LD.R</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>ENSG00000135074.15</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>157395534</td>\n",
       "      <td>157575775</td>\n",
       "      <td>R</td>\n",
       "      <td>496</td>\n",
       "      <td>chr5_157508516_C_T_b38</td>\n",
       "      <td>chr5_157517467_G_T_b38</td>\n",
       "      <td>2.140000e-69</td>\n",
       "      <td>3.940000e-66</td>\n",
       "      <td>1.410000e-05</td>\n",
       "      <td>0.024113</td>\n",
       "      <td>0.975873</td>\n",
       "      <td>0.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>ENSG00000237541.3</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>32741391</td>\n",
       "      <td>32747198</td>\n",
       "      <td>F</td>\n",
       "      <td>1760</td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>3.340000e-15</td>\n",
       "      <td>5.430000e-07</td>\n",
       "      <td>7.740000e-10</td>\n",
       "      <td>0.124045</td>\n",
       "      <td>0.875955</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FEV1</td>\n",
       "      <td>All</td>\n",
       "      <td>ENSG00000138780.14</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td>105708778</td>\n",
       "      <td>105847726</td>\n",
       "      <td>F</td>\n",
       "      <td>824</td>\n",
       "      <td>chr4_105863123_G_C_b38</td>\n",
       "      <td>chr4_105884676_C_G_b38</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>8.480000e-05</td>\n",
       "      <td>0.166543</td>\n",
       "      <td>0.833372</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     trait stratum        phenotype.id gene.name  start_pos    end_pos strand  \\\n",
       "0  FEV1FVC     All  ENSG00000135074.15    ADAM19  157395534  157575775      R   \n",
       "1  FEV1FVC     All   ENSG00000237541.3  HLA-DQA2   32741391   32747198      F   \n",
       "2     FEV1     All  ENSG00000138780.14     GSTCD  105708778  105847726      F   \n",
       "\n",
       "   nsnps             variant.id1             variant.id2  PP.H0.abf.Exam1  \\\n",
       "0    496  chr5_157508516_C_T_b38  chr5_157517467_G_T_b38     2.140000e-69   \n",
       "1   1760   chr6_31896897_T_C_b38   chr6_32183666_C_T_b38     3.340000e-15   \n",
       "2    824  chr4_105863123_G_C_b38  chr4_105884676_C_G_b38     0.000000e+00   \n",
       "\n",
       "   PP.H1.abf.Exam1  PP.H2.abf.Exam1  PP.H3.abf.Exam1  PP.H4.abf.Exam1  LD.R  \n",
       "0     3.940000e-66     1.410000e-05         0.024113         0.975873  0.99  \n",
       "1     5.430000e-07     7.740000e-10         0.124045         0.875955  0.51  \n",
       "2     0.000000e+00     8.480000e-05         0.166543         0.833372  0.98  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_df =pd.read_csv(\"/home/jve4pt/resource-files/xiaowei-analysis-files/summary_gwas_eqtl_ColocSusie_13May2022_XH.csv\")\n",
    "eqtl_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>trait</th>\n",
       "      <th>stratum</th>\n",
       "      <th>phenotype.id</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>Start_hg38</th>\n",
       "      <th>End_hg38</th>\n",
       "      <th>nsnps</th>\n",
       "      <th>variant.id1</th>\n",
       "      <th>variant.id2</th>\n",
       "      <th>...</th>\n",
       "      <th>PP.H4.abf.Exam1</th>\n",
       "      <th>PP.H0.abf.Exam5</th>\n",
       "      <th>PP.H1.abf.Exam5</th>\n",
       "      <th>PP.H2.abf.Exam5</th>\n",
       "      <th>PP.H3.abf.Exam5</th>\n",
       "      <th>PP.H4.abf.Exam5</th>\n",
       "      <th>LD.R</th>\n",
       "      <th>UCSC_RefGene_Name</th>\n",
       "      <th>UCSC_RefGene_Group</th>\n",
       "      <th>Relation_to_UCSC_CpG_Island</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>cg07318204</td>\n",
       "      <td>4</td>\n",
       "      <td>145566441</td>\n",
       "      <td>144645288</td>\n",
       "      <td>144645290</td>\n",
       "      <td>617</td>\n",
       "      <td>chr4_144541436_A_C_b38</td>\n",
       "      <td>chr4_144548816_G_T_b38</td>\n",
       "      <td>...</td>\n",
       "      <td>0.949431</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>HHIP</td>\n",
       "      <td>TSS1500</td>\n",
       "      <td>Island</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>cg10728227</td>\n",
       "      <td>4</td>\n",
       "      <td>145478755</td>\n",
       "      <td>144557602</td>\n",
       "      <td>144557604</td>\n",
       "      <td>617</td>\n",
       "      <td>chr4_144541436_A_C_b38</td>\n",
       "      <td>chr4_144557510_C_T_b38</td>\n",
       "      <td>...</td>\n",
       "      <td>0.963304</td>\n",
       "      <td>3.840000e-44</td>\n",
       "      <td>2.250000e-36</td>\n",
       "      <td>6.620000e-10</td>\n",
       "      <td>0.036781</td>\n",
       "      <td>0.963219</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>cg13749822</td>\n",
       "      <td>4</td>\n",
       "      <td>145566663</td>\n",
       "      <td>144645510</td>\n",
       "      <td>144645512</td>\n",
       "      <td>617</td>\n",
       "      <td>chr4_144541436_A_C_b38</td>\n",
       "      <td>chr4_144548816_G_T_b38</td>\n",
       "      <td>...</td>\n",
       "      <td>0.975014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>HHIP</td>\n",
       "      <td>TSS1500</td>\n",
       "      <td>Island</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>cg26339943</td>\n",
       "      <td>4</td>\n",
       "      <td>145571234</td>\n",
       "      <td>144650081</td>\n",
       "      <td>144650083</td>\n",
       "      <td>617</td>\n",
       "      <td>chr4_144541436_A_C_b38</td>\n",
       "      <td>chr4_144550093_A_T_b38</td>\n",
       "      <td>...</td>\n",
       "      <td>0.933779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.95</td>\n",
       "      <td>HHIP</td>\n",
       "      <td>Body</td>\n",
       "      <td>S_Shelf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>cg01197005</td>\n",
       "      <td>5</td>\n",
       "      <td>147677420</td>\n",
       "      <td>148297856</td>\n",
       "      <td>148297858</td>\n",
       "      <td>1117</td>\n",
       "      <td>chr5_148360432_A_G_b38</td>\n",
       "      <td>chr5_148355813_T_C_b38</td>\n",
       "      <td>...</td>\n",
       "      <td>0.952336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>LOC102546294</td>\n",
       "      <td>Body</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     trait stratum phenotype.id  chr   pos_hg19  Start_hg38   End_hg38  nsnps  \\\n",
       "0  FEV1FVC     All   cg07318204    4  145566441   144645288  144645290    617   \n",
       "1  FEV1FVC     All   cg10728227    4  145478755   144557602  144557604    617   \n",
       "2  FEV1FVC     All   cg13749822    4  145566663   144645510  144645512    617   \n",
       "3  FEV1FVC     All   cg26339943    4  145571234   144650081  144650083    617   \n",
       "4  FEV1FVC     All   cg01197005    5  147677420   148297856  148297858   1117   \n",
       "\n",
       "              variant.id1             variant.id2  ...  PP.H4.abf.Exam1  \\\n",
       "0  chr4_144541436_A_C_b38  chr4_144548816_G_T_b38  ...         0.949431   \n",
       "1  chr4_144541436_A_C_b38  chr4_144557510_C_T_b38  ...         0.963304   \n",
       "2  chr4_144541436_A_C_b38  chr4_144548816_G_T_b38  ...         0.975014   \n",
       "3  chr4_144541436_A_C_b38  chr4_144550093_A_T_b38  ...         0.933779   \n",
       "4  chr5_148360432_A_G_b38  chr5_148355813_T_C_b38  ...         0.952336   \n",
       "\n",
       "   PP.H0.abf.Exam5  PP.H1.abf.Exam5  PP.H2.abf.Exam5  PP.H3.abf.Exam5  \\\n",
       "0              NaN              NaN              NaN              NaN   \n",
       "1     3.840000e-44     2.250000e-36     6.620000e-10         0.036781   \n",
       "2              NaN              NaN              NaN              NaN   \n",
       "3              NaN              NaN              NaN              NaN   \n",
       "4              NaN              NaN              NaN              NaN   \n",
       "\n",
       "   PP.H4.abf.Exam5  LD.R  UCSC_RefGene_Name  UCSC_RefGene_Group  \\\n",
       "0              NaN  0.99               HHIP             TSS1500   \n",
       "1         0.963219  0.99                NaN                 NaN   \n",
       "2              NaN  0.99               HHIP             TSS1500   \n",
       "3              NaN  0.95               HHIP                Body   \n",
       "4              NaN  1.00       LOC102546294                Body   \n",
       "\n",
       "   Relation_to_UCSC_CpG_Island  \n",
       "0                       Island  \n",
       "1                          NaN  \n",
       "2                       Island  \n",
       "3                      S_Shelf  \n",
       "4                          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mqtl_df =pd.read_csv(\"/home/jve4pt/resource-files/xiaowei-analysis-files/summary_gwas_mqtl_ColocSusie_13May2022_XH.csv\")\n",
    "mqtl_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>trait</th>\n",
       "      <th>stratum</th>\n",
       "      <th>phenotype.id</th>\n",
       "      <th>EntrezGeneSymbol</th>\n",
       "      <th>rsid1</th>\n",
       "      <th>rsid2</th>\n",
       "      <th>nsnps</th>\n",
       "      <th>variant.id1</th>\n",
       "      <th>variant.id2</th>\n",
       "      <th>PP.H0.abf</th>\n",
       "      <th>PP.H1.abf</th>\n",
       "      <th>PP.H2.abf</th>\n",
       "      <th>PP.H3.abf</th>\n",
       "      <th>PP.H4.abf</th>\n",
       "      <th>LD.R</th>\n",
       "      <th>anno.rsid1</th>\n",
       "      <th>anno.rsid2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FEV1FVC</td>\n",
       "      <td>All</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>AGER</td>\n",
       "      <td>rs2844456</td>\n",
       "      <td>rs9391855</td>\n",
       "      <td>1477</td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td>5.540000e-13</td>\n",
       "      <td>0.000287</td>\n",
       "      <td>1.320000e-10</td>\n",
       "      <td>0.066289</td>\n",
       "      <td>0.933424</td>\n",
       "      <td>0.58</td>\n",
       "      <td>EHMT2_Intron; C2_2KB Upstream</td>\n",
       "      <td>AGER_Intron</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     trait stratum                 phenotype.id EntrezGeneSymbol      rsid1  \\\n",
       "0  FEV1FVC     All  SL003680_ENSG00000204305.14             AGER  rs2844456   \n",
       "\n",
       "       rsid2  nsnps            variant.id1            variant.id2  \\\n",
       "0  rs9391855   1477  chr6_31896897_T_C_b38  chr6_32182024_C_T_b38   \n",
       "\n",
       "      PP.H0.abf  PP.H1.abf     PP.H2.abf  PP.H3.abf  PP.H4.abf  LD.R  \\\n",
       "0  5.540000e-13   0.000287  1.320000e-10   0.066289   0.933424  0.58   \n",
       "\n",
       "                      anno.rsid1   anno.rsid2  \n",
       "0  EHMT2_Intron; C2_2KB Upstream  AGER_Intron  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pqtl_df =pd.read_csv(\"/home/jve4pt/resource-files/xiaowei-analysis-files/summary_gwas_pqtl_ColocSusie_27May2022_XH.csv\")\n",
    "pqtl_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract Unique Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr5_157508516_C_T_b38',\n",
       " 'chr4_105884676_C_G_b38',\n",
       " 'chr5_157517467_G_T_b38',\n",
       " 'chr6_32183666_C_T_b38',\n",
       " 'chr4_105863123_G_C_b38',\n",
       " 'chr6_31896897_T_C_b38']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mqtl=list(set(mqtl_df[\"variant.id1\"].tolist() + mqtl_df[\"variant.id2\"].tolist()))\n",
    "pqtl=list(set(pqtl_df[\"variant.id1\"].tolist() + pqtl_df[\"variant.id2\"].tolist()))\n",
    "eqtl=list(set(eqtl_df[\"variant.id1\"].tolist() + eqtl_df[\"variant.id2\"].tolist()))\n",
    "\n",
    "eqtl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fuzzy & Exact Matching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "distance_threshold = 1000000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Expression and Methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp_df = pd.DataFrame()\n",
    "\n",
    "for eqtl_variant in eqtl:\n",
    "    for mqtl_variant in mqtl:\n",
    "        # if on the same chromosome \n",
    "        if eqtl_variant.split(\"_\")[0] == mqtl_variant.split(\"_\")[0]:\n",
    "            distance = abs(int(eqtl_variant.split(\"_\")[1])-int(mqtl_variant.split(\"_\")[1]))\n",
    "            # if they're close enough \n",
    "            if distance <= distance_threshold:\n",
    "\n",
    "                tmp_df = pd.concat(\n",
    "                    [tmp_df,\n",
    "                         pd.DataFrame(\n",
    "                             [[eqtl_variant,\n",
    "                               # get all expression genes for variants in expression matrix and join them by comma\n",
    "                               \",\".join(eqtl_df[(eqtl_df[\"variant.id1\"] == eqtl_variant) | (eqtl_df[\"variant.id2\"] == eqtl_variant)][\"gene.name\"].tolist()),\n",
    "                               # same as above but also look at methylation sheet for an eqtl in case it's also an mqtl\n",
    "                               \",\".join(mqtl_df[(mqtl_df[\"variant.id1\"] == eqtl_variant) | (mqtl_df[\"variant.id2\"] == eqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               mqtl_variant,\n",
    "                               # mqtl in eqtl sheet \n",
    "                               \",\".join(eqtl_df[(eqtl_df[\"variant.id1\"] == mqtl_variant) | (eqtl_df[\"variant.id2\"] == mqtl_variant)][\"gene.name\"].tolist()),\n",
    "                               # mqtl in  mqtl sheet \n",
    "                               \",\".join(mqtl_df[(mqtl_df[\"variant.id1\"] == mqtl_variant) | (mqtl_df[\"variant.id2\"] == mqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               distance\n",
    "                              ]]\n",
    "                         )\n",
    "                    ]\n",
    "                )\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Expression Variant</th>\n",
       "      <th>Expression Variant Expression Gene</th>\n",
       "      <th>Expression Variant CpG Site</th>\n",
       "      <th>Methylation Variant</th>\n",
       "      <th>Methylation Variant Expression Gene</th>\n",
       "      <th>Methylation Variant CpG Site</th>\n",
       "      <th>Distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_157508516_C_T_b38</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>cg25090510</td>\n",
       "      <td>chr5_157508516_C_T_b38</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>cg25090510</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_105884676_C_G_b38</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td>cg19338966</td>\n",
       "      <td>chr4_105884676_C_G_b38</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td>cg19338966</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_105884676_C_G_b38</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td>cg19338966</td>\n",
       "      <td>chr4_105882525_A_C_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg19338966</td>\n",
       "      <td>2151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5_157517467_G_T_b38</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td></td>\n",
       "      <td>chr5_157508516_C_T_b38</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>cg25090510</td>\n",
       "      <td>8951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_105863123_G_C_b38</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td></td>\n",
       "      <td>chr4_105882525_A_C_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg19338966</td>\n",
       "      <td>19402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4_105863123_G_C_b38</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td></td>\n",
       "      <td>chr4_105884676_C_G_b38</td>\n",
       "      <td>GSTCD</td>\n",
       "      <td>cg19338966</td>\n",
       "      <td>21553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_32053184_G_C_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg07810190,cg26997880,cg07810190,cg26997880</td>\n",
       "      <td>130482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>chr6_32053184_G_C_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg07810190,cg26997880,cg07810190,cg26997880</td>\n",
       "      <td>156287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>chr6_31714252_G_A_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg05483184,cg22547540,cg04960128,cg05483184,cg...</td>\n",
       "      <td>182645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>chr6_31636578_C_A_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg15316118,cg05483184,cg07810190,cg10812186,cg...</td>\n",
       "      <td>260319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>chr6_31616064_C_G_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg15316118,cg10812186,cg20471413,cg24055029,cg...</td>\n",
       "      <td>280833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>286769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31714252_G_A_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg05483184,cg22547540,cg04960128,cg05483184,cg...</td>\n",
       "      <td>469414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31636578_C_A_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg15316118,cg05483184,cg07810190,cg10812186,cg...</td>\n",
       "      <td>547088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31616064_C_G_b38</td>\n",
       "      <td></td>\n",
       "      <td>cg15316118,cg10812186,cg20471413,cg24055029,cg...</td>\n",
       "      <td>567602</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Expression Variant Expression Variant Expression Gene  \\\n",
       "0  chr5_157508516_C_T_b38                             ADAM19   \n",
       "0  chr4_105884676_C_G_b38                              GSTCD   \n",
       "0   chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0  chr4_105884676_C_G_b38                              GSTCD   \n",
       "0  chr5_157517467_G_T_b38                             ADAM19   \n",
       "0  chr4_105863123_G_C_b38                              GSTCD   \n",
       "0  chr4_105863123_G_C_b38                              GSTCD   \n",
       "0   chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "0   chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0   chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0   chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0   chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0   chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "0   chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "0   chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "0   chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "\n",
       "                         Expression Variant CpG Site     Methylation Variant  \\\n",
       "0                                         cg25090510  chr5_157508516_C_T_b38   \n",
       "0                                         cg19338966  chr4_105884676_C_G_b38   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   chr6_31896897_T_C_b38   \n",
       "0                                         cg19338966  chr4_105882525_A_C_b38   \n",
       "0                                                     chr5_157508516_C_T_b38   \n",
       "0                                                     chr4_105882525_A_C_b38   \n",
       "0                                                     chr4_105884676_C_G_b38   \n",
       "0                                                      chr6_32053184_G_C_b38   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   chr6_32053184_G_C_b38   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   chr6_31714252_G_A_b38   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   chr6_31636578_C_A_b38   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   chr6_31616064_C_G_b38   \n",
       "0                                                      chr6_31896897_T_C_b38   \n",
       "0                                                      chr6_31714252_G_A_b38   \n",
       "0                                                      chr6_31636578_C_A_b38   \n",
       "0                                                      chr6_31616064_C_G_b38   \n",
       "\n",
       "  Methylation Variant Expression Gene  \\\n",
       "0                              ADAM19   \n",
       "0                               GSTCD   \n",
       "0                            HLA-DQA2   \n",
       "0                                       \n",
       "0                              ADAM19   \n",
       "0                                       \n",
       "0                               GSTCD   \n",
       "0                                       \n",
       "0                                       \n",
       "0                                       \n",
       "0                                       \n",
       "0                                       \n",
       "0                            HLA-DQA2   \n",
       "0                                       \n",
       "0                                       \n",
       "0                                       \n",
       "\n",
       "                        Methylation Variant CpG Site  Distance  \n",
       "0                                         cg25090510         0  \n",
       "0                                         cg19338966         0  \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...         0  \n",
       "0                                         cg19338966      2151  \n",
       "0                                         cg25090510      8951  \n",
       "0                                         cg19338966     19402  \n",
       "0                                         cg19338966     21553  \n",
       "0        cg07810190,cg26997880,cg07810190,cg26997880    130482  \n",
       "0        cg07810190,cg26997880,cg07810190,cg26997880    156287  \n",
       "0  cg05483184,cg22547540,cg04960128,cg05483184,cg...    182645  \n",
       "0  cg15316118,cg05483184,cg07810190,cg10812186,cg...    260319  \n",
       "0  cg15316118,cg10812186,cg20471413,cg24055029,cg...    280833  \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...    286769  \n",
       "0  cg05483184,cg22547540,cg04960128,cg05483184,cg...    469414  \n",
       "0  cg15316118,cg05483184,cg07810190,cg10812186,cg...    547088  \n",
       "0  cg15316118,cg10812186,cg20471413,cg24055029,cg...    567602  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename columns\n",
    "tmp_df.columns=[\n",
    "    \"Expression Variant\", \n",
    "    \"Expression Variant Expression Gene\", \n",
    "    \"Expression Variant CpG Site\", \n",
    "    \"Methylation Variant\", \n",
    "    \"Methylation Variant Expression Gene\", \n",
    "    \"Methylation Variant CpG Site\",\n",
    "    \"Distance\"\n",
    "]\n",
    "\n",
    "tmp_df.sort_values(\"Distance\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# net.draw(net.from_pandas_edgelist(tmp_df, \"Expression\",\"Methylation\"), with_labels=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Expression and Protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp_df = pd.DataFrame()\n",
    "\n",
    "for eqtl_variant in eqtl:\n",
    "    for pqtl_variant in pqtl:\n",
    "        # if on the same chromosome \n",
    "        if eqtl_variant.split(\"_\")[0] == pqtl_variant.split(\"_\")[0]:\n",
    "            distance = abs(int(eqtl_variant.split(\"_\")[1])-int(pqtl_variant.split(\"_\")[1]))\n",
    "            # if they're close enough \n",
    "            if distance <= distance_threshold:\n",
    "\n",
    "                tmp_df = pd.concat(\n",
    "                    [tmp_df,\n",
    "                         pd.DataFrame(\n",
    "                             [[eqtl_variant,\n",
    "                               # get all expression genes for variants in expression matrix and join them by comma\n",
    "                               \",\".join(eqtl_df[(eqtl_df[\"variant.id1\"] == eqtl_variant) | (eqtl_df[\"variant.id2\"] == eqtl_variant)][\"gene.name\"].tolist()),\n",
    "                               # same as above but also look at methylation sheet for an eqtl in case it's also an pqtl\n",
    "                               \",\".join(pqtl_df[(pqtl_df[\"variant.id1\"] == eqtl_variant) | (pqtl_df[\"variant.id2\"] == eqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               pqtl_variant,\n",
    "                               # pqtl in eqtl sheet \n",
    "                               \",\".join(eqtl_df[(eqtl_df[\"variant.id1\"] == pqtl_variant) | (eqtl_df[\"variant.id2\"] == pqtl_variant)][\"gene.name\"].tolist()),\n",
    "                               # pqtl in  pqtl sheet \n",
    "                               \",\".join(pqtl_df[(pqtl_df[\"variant.id1\"] == pqtl_variant) | (pqtl_df[\"variant.id2\"] == pqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               distance\n",
    "                              ]]\n",
    "                         )\n",
    "                    ]\n",
    "                )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Expression Variant</th>\n",
       "      <th>Expression Variant Expression Gene</th>\n",
       "      <th>Expression Variant Protein</th>\n",
       "      <th>Protein Variant</th>\n",
       "      <th>Protein Variant Expression Gene</th>\n",
       "      <th>Protein Variant Protein</th>\n",
       "      <th>Distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>1642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>285127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32183666_C_T_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>HLA-DQA2</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>286769</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Expression Variant Expression Variant Expression Gene  \\\n",
       "0  chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0  chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "0  chr6_31896897_T_C_b38                           HLA-DQA2   \n",
       "0  chr6_32183666_C_T_b38                           HLA-DQA2   \n",
       "\n",
       "    Expression Variant Protein        Protein Variant  \\\n",
       "0  SL003680_ENSG00000204305.14  chr6_31896897_T_C_b38   \n",
       "0                               chr6_32182024_C_T_b38   \n",
       "0  SL003680_ENSG00000204305.14  chr6_32182024_C_T_b38   \n",
       "0                               chr6_31896897_T_C_b38   \n",
       "\n",
       "  Protein Variant Expression Gene      Protein Variant Protein  Distance  \n",
       "0                        HLA-DQA2  SL003680_ENSG00000204305.14         0  \n",
       "0                                  SL003680_ENSG00000204305.14      1642  \n",
       "0                                  SL003680_ENSG00000204305.14    285127  \n",
       "0                        HLA-DQA2  SL003680_ENSG00000204305.14    286769  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename columns\n",
    "tmp_df.columns=[\n",
    "    \"Expression Variant\", \n",
    "    \"Expression Variant Expression Gene\", \n",
    "    \"Expression Variant Protein\", \n",
    "    \"Protein Variant\", \n",
    "    \"Protein Variant Expression Gene\", \n",
    "    \"Protein Variant Protein\",\n",
    "    \"Distance\"\n",
    "]\n",
    "\n",
    "tmp_df.sort_values(\"Distance\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Methylation and Protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp_df = pd.DataFrame()\n",
    "\n",
    "for mqtl_variant in mqtl:\n",
    "    for pqtl_variant in pqtl:\n",
    "        # if on the same chromosome \n",
    "        if mqtl_variant.split(\"_\")[0] == pqtl_variant.split(\"_\")[0]:\n",
    "            distance = abs(int(mqtl_variant.split(\"_\")[1])-int(pqtl_variant.split(\"_\")[1]))\n",
    "            # if they're close enough \n",
    "            if distance <= distance_threshold:\n",
    "\n",
    "                tmp_df = pd.concat(\n",
    "                    [tmp_df,\n",
    "                         pd.DataFrame(\n",
    "                             [[mqtl_variant,\n",
    "                               # get all expression genes for variants in expression matrix and join them by comma\n",
    "                               \",\".join(mqtl_df[(mqtl_df[\"variant.id1\"] == mqtl_variant) | (mqtl_df[\"variant.id2\"] == mqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               # same as above but also look at methylation sheet for an mqtl in case it's also an pqtl\n",
    "                               \",\".join(pqtl_df[(pqtl_df[\"variant.id1\"] == mqtl_variant) | (pqtl_df[\"variant.id2\"] == mqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               pqtl_variant,\n",
    "                               # pqtl in mqtl sheet \n",
    "                               \",\".join(mqtl_df[(mqtl_df[\"variant.id1\"] == pqtl_variant) | (mqtl_df[\"variant.id2\"] == pqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               # pqtl in  pqtl sheet \n",
    "                               \",\".join(pqtl_df[(pqtl_df[\"variant.id1\"] == pqtl_variant) | (pqtl_df[\"variant.id2\"] == pqtl_variant)][\"phenotype.id\"].tolist()),\n",
    "                               distance\n",
    "                              ]]\n",
    "                         )\n",
    "                    ]\n",
    "                )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Methylation Variant</th>\n",
       "      <th>Methylation Variant CpG Site</th>\n",
       "      <th>Methylation Variant Protein</th>\n",
       "      <th>Protein Variant</th>\n",
       "      <th>Protein Variant CpG Site</th>\n",
       "      <th>Protein Variant Protein</th>\n",
       "      <th>Distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32053184_G_C_b38</td>\n",
       "      <td>cg07810190,cg26997880,cg07810190,cg26997880</td>\n",
       "      <td></td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>128840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_32053184_G_C_b38</td>\n",
       "      <td>cg07810190,cg26997880,cg07810190,cg26997880</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>156287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31714252_G_A_b38</td>\n",
       "      <td>cg05483184,cg22547540,cg04960128,cg05483184,cg...</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>182645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31636578_C_A_b38</td>\n",
       "      <td>cg15316118,cg05483184,cg07810190,cg10812186,cg...</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>260319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31616064_C_G_b38</td>\n",
       "      <td>cg15316118,cg10812186,cg20471413,cg24055029,cg...</td>\n",
       "      <td></td>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>280833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31896897_T_C_b38</td>\n",
       "      <td>cg05483184,cg07810190,cg10812186,cg20471413,cg...</td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>285127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31714252_G_A_b38</td>\n",
       "      <td>cg05483184,cg22547540,cg04960128,cg05483184,cg...</td>\n",
       "      <td></td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>467772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31636578_C_A_b38</td>\n",
       "      <td>cg15316118,cg05483184,cg07810190,cg10812186,cg...</td>\n",
       "      <td></td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>545446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_31616064_C_G_b38</td>\n",
       "      <td>cg15316118,cg10812186,cg20471413,cg24055029,cg...</td>\n",
       "      <td></td>\n",
       "      <td>chr6_32182024_C_T_b38</td>\n",
       "      <td></td>\n",
       "      <td>SL003680_ENSG00000204305.14</td>\n",
       "      <td>565960</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Methylation Variant                       Methylation Variant CpG Site  \\\n",
       "0  chr6_31896897_T_C_b38  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0  chr6_32053184_G_C_b38        cg07810190,cg26997880,cg07810190,cg26997880   \n",
       "0  chr6_32053184_G_C_b38        cg07810190,cg26997880,cg07810190,cg26997880   \n",
       "0  chr6_31714252_G_A_b38  cg05483184,cg22547540,cg04960128,cg05483184,cg...   \n",
       "0  chr6_31636578_C_A_b38  cg15316118,cg05483184,cg07810190,cg10812186,cg...   \n",
       "0  chr6_31616064_C_G_b38  cg15316118,cg10812186,cg20471413,cg24055029,cg...   \n",
       "0  chr6_31896897_T_C_b38  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0  chr6_31714252_G_A_b38  cg05483184,cg22547540,cg04960128,cg05483184,cg...   \n",
       "0  chr6_31636578_C_A_b38  cg15316118,cg05483184,cg07810190,cg10812186,cg...   \n",
       "0  chr6_31616064_C_G_b38  cg15316118,cg10812186,cg20471413,cg24055029,cg...   \n",
       "\n",
       "   Methylation Variant Protein        Protein Variant  \\\n",
       "0  SL003680_ENSG00000204305.14  chr6_31896897_T_C_b38   \n",
       "0                               chr6_32182024_C_T_b38   \n",
       "0                               chr6_31896897_T_C_b38   \n",
       "0                               chr6_31896897_T_C_b38   \n",
       "0                               chr6_31896897_T_C_b38   \n",
       "0                               chr6_31896897_T_C_b38   \n",
       "0  SL003680_ENSG00000204305.14  chr6_32182024_C_T_b38   \n",
       "0                               chr6_32182024_C_T_b38   \n",
       "0                               chr6_32182024_C_T_b38   \n",
       "0                               chr6_32182024_C_T_b38   \n",
       "\n",
       "                            Protein Variant CpG Site  \\\n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0                                                      \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0  cg05483184,cg07810190,cg10812186,cg20471413,cg...   \n",
       "0                                                      \n",
       "0                                                      \n",
       "0                                                      \n",
       "0                                                      \n",
       "\n",
       "       Protein Variant Protein  Distance  \n",
       "0  SL003680_ENSG00000204305.14         0  \n",
       "0  SL003680_ENSG00000204305.14    128840  \n",
       "0  SL003680_ENSG00000204305.14    156287  \n",
       "0  SL003680_ENSG00000204305.14    182645  \n",
       "0  SL003680_ENSG00000204305.14    260319  \n",
       "0  SL003680_ENSG00000204305.14    280833  \n",
       "0  SL003680_ENSG00000204305.14    285127  \n",
       "0  SL003680_ENSG00000204305.14    467772  \n",
       "0  SL003680_ENSG00000204305.14    545446  \n",
       "0  SL003680_ENSG00000204305.14    565960  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename columns\n",
    "tmp_df.columns=[\n",
    "    \"Methylation Variant\", \n",
    "    \"Methylation Variant CpG Site\", \n",
    "    \"Methylation Variant Protein\", \n",
    "    \"Protein Variant\", \n",
    "    \"Protein Variant CpG Site\", \n",
    "    \"Protein Variant Protein\",\n",
    "    \"Distance\"\n",
    "]\n",
    "\n",
    "tmp_df.sort_values(\"Distance\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8",
   "language": "python",
   "name": "python-3.8"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  },
  "vscode": {
   "interpreter": {
    "hash": "ce68a6b3d2c1fa73c998106b6f5782b55a91eabf989771c458cf16c331825b05"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
